Sources/MessagingInApp/Gist/Gist.swift

```diff
@@ -83,13 +85,24 @@ public class Gist: GistDelegate {
     // MARK: Events
 
     public func messageShown(message: Message) {
+        let queueId = message.queueId
+        
+        if queueId != nil && shownMessageQueueIds.contains(queueId!) {
```

Comment:
I think we probably need to check for if the message is `persistent` or not, @BernardGatt correct me if i am wrong here. 

Because with persistent we load the same message until dismissed?

Reply:
Good catch Shahroz!!

To avoid this we could check if the message is persistent and if it is we don't add it to the list. We only add it when dismissMessage is called and the message is persistent.

What do you think?

Reply:
Even better !!